src/lib.rs

```diff
@@ -59,6 +42,24 @@
 //! let solution = resolve(&dependency_provider, "root", 1u32).unwrap();
 //! ```
 //!
+//! # Package and Version flexibility
+//!
+//! The [OfflineDependencyProvider] used in that example is generic over the way package names,
+//! version requirements, and version numbers are represented.
+//!
+//! The first bound is the type of package names. It can be anything that implements our [Package] trait.
+//! The [Package] trait is automatic if the type already implements
+//! [Clone] + [Eq] + [Hash] + [Debug] + [Display](std::fmt::Display).
```

Comment:
Why's the `Display` link different than the others?

Reply:
Because the others are in the prelude, and `Display` is not.